Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs

```diff
@@ -26,12 +27,28 @@ public T GetValue<T>(string key, T defaultValue)
                 var found = _current.TryGetValue(key, out result);
                 if (found)
                 {
+                    // This is a workaround for a bug on UWP where Install-Id cannot be read 
+                    // from Unity.PlayerPrefs, because it is saved as System.Guid, but read as a string.
+                    if ((typeof(T) == typeof(Guid?)) && (result is string))
+                    {
+                        try 
+                        {
+                            object guid = (object)(new Guid((string)result));
+                            return (T)guid;
+                        }
+                        catch (FormatException)
+                        {
+                            Debug.LogError("Guid cannot be parsed due to a format exception.");
+                            return defaultValue;
+                        }
```

Comment:
```suggestion
                        if (Guid.TryParse((string?) result, out var guid)) {
                            return guid.ToString();
                        }
                       else
                       {
                            Debug.LogError("Guid cannot be parsed due to a format exception.");
                            return defaultValue;
                        }
```

Reply:
It won't compile: to make it compile we need a double cast